### ISSUE TITLE
[BUG]Msg duplicate when consumer in Failover mode

### ISSUE THREAD

**Issue by archfish** [apache/pulsar#6535]
**Describe the bug**
Msg duplicate when consumer in Failover mode

**To Reproduce**
1. create a subscribe name A to topic B
2. send some message C to topic B
3. create consumer with subscribe name A in Failover mode, consumer name AA
4. get message C in consumer AA, an sleep seconds
5. create consumer with subscribe name A in Failover mode, consumer name BB
6.  consumer BB get message C

**Expected behavior**

- Do not change active consumer (maybe best)

- Or do not deliver message which already delivered to other consumer

**Screenshots**

![image](https://user-images.githubusercontent.com/5317064/76604476-bc09d380-6549-11ea-9049-032e48ca3680.png)

**Comment by archfish** [comment]
```golang
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"github.com/apache/pulsar-client-go/pulsar"
)

func main() {
	client, err := pulsar.NewClient(pulsar.ClientOptions{URL: "pulsar://my.pulsar.lan:6650"})
	if err != nil {
		log.Fatal(err)
	}

	defer client.Close()

	consumer, err := client.Subscribe(pulsar.ConsumerOptions{
		Topic:            "shared",
		SubscriptionName: "my-sub",
		Type:             pulsar.Failover,
	})
	if err != nil {
		log.Fatal(err)
	}
	defer consumer.Close()

	for i := 0; i < 10; i++ {
		msg, err := consumer.Receive(context.Background())
		if err != nil {
			log.Fatal(err)
		}

		fmt.Printf("Received message msgId: %#v -- content: '%s'\n",
			msg.ID(), string(msg.Payload()))

		// consumer.Ack(msg)
		time.Sleep(10 * time.Second)
	}

	if err := consumer.Unsubscribe(); err != nil {
		log.Fatal(err)
	}
}
```

```shell
# at least has one message in topic
go run examples/consumer/consumer.go
# after consumer started
go run examples/consumer/consumer.go
```

**Comment by jiazhai** [comment]
hi @archfish  There maybe some mis-understanding regarding the dup.  Only an ack-ed message is considered no need for redeliver in pulsar. 
After #4604, included in 2.5.0 release, only the first connected consumer is active, it should avoid the consumer switch.

**Comment by archfish** [comment]
> hi @archfish There maybe some mis-understanding regarding the dup. Only an ack-ed message is considered no need for redeliver in pulsar.
> After #4604, included in 2.5.0 release, only the first connected consumer is active, it should avoid the consumer switch.

This helps, Thanks!
